File: Algorithm/insertion_sort/insertion-sort.py
import time
import random
start_time = time.time()
def insertionSort(element):
    for i in range(1, len(element)):
        anchor = element[i]
        j = i-1
        while j >= 0 and anchor < element[j]:
            element[j + 1] = element[j]
            j = j -1
        element[j+1] = anchor

def run_median(stream):
    arr = []

    for i in stream:
        arr.append(i)
        insertionSort(arr)

        n = len(arr)
        if n % 2 == 1:
            median = arr[n // 2]
        else:
            median = (arr[n // 2] + arr[n // 2 - 1]) / 2
        print(median)



needToSorted = [random.randint(1, 10000) for _ in range(10000)] 
# insertionSort(needToSorted)
run_median(needToSorted)
print(needToSorted)

end_time = time.time()
duration = end_time - start_time

print(f"waktu yang di perlukan: ${duration} ")

File: Algorithm/insertion_sort/insertion-sort.cpp
#include <iostream>
using namespace std;

void insertion_sort(int arrays[],int n);
int main(){
    int arrays[] = {10,2,4,6,8,9,7,5,3,1};
    int n = sizeof(arrays)/sizeof(arrays[0]);

    insertion_sort(arrays,n);

    for(int i = 0; i < n; i ++){
        cout << arrays[i] << " ";
    }
}

void insertion_sort(int arrays[],int n){
    for(int i = 1; i < n; i++){
        int anchor = arrays[i];
        int j = i-1;
        while  (j >= 0 && anchor < arrays[j]){
            arrays[j+1] = arrays[j];
            j = j - 1;
        }
        arrays[j + 1] = anchor;
    }
}